packages/deployer/src/deployer.ts

```diff
@@ -49,6 +49,18 @@ export class Deployer {
             gas,
         };
         const abi = contractNetworkDataIfExists.abi;
+        const constructorAbi = _.find(abi, { type: AbiType.Constructor }) as Web3.ConstructorAbi;
+        const constructorArgs = _.isUndefined(constructorAbi) ? [] : constructorAbi.inputs;
+        if (constructorArgs.length !== args.length) {
+            const constructorSignature = `constructor(${_.map(constructorArgs, arg => `${arg.type} ${arg.name}`).join(
```

Comment:
Mh, not the biggest fan of embedding interators *inside* of template strings. Can we move it to the line above for readability?